### PR TITLE
Add false as possible type in styled props

### PR DESCRIPTION
**Pull request by devrnt**
### Problem
```tsx
const Box = styled('div')<{ border: boolean }>`
   color: red; 

   ${({ border }) =>
       border && `
         border: 1px solid red;
      `
  }
`;
```

#### TSC output: 
```
...
Type 'false' is not assignable to type 'string | number | CSSAttribute | undefined'.
```

### Current workaround 
Use ternary operator and return empty string for the falsy value

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

### CODE REVIEW

File: src/__tests__/styled.test.js [hunk 1]
```diff
@@ -146,4 +146,14 @@ describe('styled', () => {
 
         expect(vnode).toMatchVNode('tag', { className: 'go', bar: 1 });
     });
+
+    it('pass truthy logical and operator', () => {
```
Comment (devrnt) [inline src/__tests__/styled.test.js:150]: Actually can be dropped since it's kind of verbose. I just added this here to verify the `&&` operator

Comment (cristianbote) [reply src/__tests__/styled.test.js:150]: But this test does not run with TypeScript 🤔  does it surface the same problem in JavaScript as well?

Comment (devrnt) [reply src/__tests__/styled.test.js:150]: No, it's working like expected. In general this "issue" can be solved by just ignoring the tsc on the user's end (aka adding `@ts-ignore`)

Comment (cristianbote) [reply src/__tests__/styled.test.js:150]: Oh, I see! 😄  Do you think we can capture it in here instead? https://github.com/cristianbote/goober/blob/master/ts-tests/test-api.tsx

Comment (devrnt) [reply src/__tests__/styled.test.js:150]: Okay I added them there. Undoing the `global.d.ts` update of this PR will make the TSC "complain" for the `<BoxWithLogicalAnd />`

**Review by cristianbote — APPROVED** [review]
👍  looks great!

**Comment by cristianbote** [comment]
Published `2.0.25` with this fix

**Pull request merged**